### ISSUE TITLE
async_without_await false positive for computed property returning optional async function

### ISSUE THREAD

**Issue by rberggreen** [realm/SwiftLint#6253]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Steps to reproduce:

Enable the async_without_await rule.  

Write a computed property that returns an optional `async` function. Implement the computed property using `Optional.map`, mapping one optional function to another. Ensure that the function returned actually uses `await` somewhere. 

Expected:

Since this computed property returns a function that uses `await`, there should be no linter warnings. 

Actual:

An async_without_await linter warning is raised.

No warnings are raised for computed properties or methods producing optional async functions if they are written in other styles, e.g. using guard instead of Optional.map. 

```swift
class SomeFactory {
    private let someInjectedValueProducer: (() async -> Int)?

    init(someInjectedValueProducer: (() async -> Int)?) {
        self.someInjectedValueProducer = someInjectedValueProducer
    }

    var someOptionalFunction: ((Int) async throws -> Int)? { // async_without_await triggered
        someInjectedValueProducer.map { someInjectedValueProducer in
            { value in
                 await someInjectedValueProducer() + 1
            }
        }
    }
}
```

<img width="1112" height="1184" alt="Image" src="https://github.com/user-attachments/assets/36a55fe2-457c-4eb2-bec1-074a516a24a9" />

### Environment

* SwiftLint version: 0.61.0
* Xcode version: 26.0 (17A324)
* Installation method used: SPM plugin from https://github.com/SimplyDanny/SwiftLintPlugins 
* Configuration file: 

```yml
only_rules:
  - async_without_await
```
